### PR TITLE
Initial Docker Setup

### PR DESCRIPTION
**Pull request by lxcodes**
This needs documentation bad! But let's get this pull request started to try some of the ideas out and see if they are worth the effort to expand further =]

I'll get setup instructions out shortly for people to try.

This is related to issue #15.

### CODE REVIEW

**Comment by htmelvis** [comment]
Twas getting going on this and was stopped by kitematic. Too late now will continue tomorrow. 
screen shot 2015-07-16 at 12 53 13 am

**Comment by lxcodes** [comment]
@htmelvis Dude, you aren't supposed to break it and make me look bad. >.<

**Comment by htmelvis** [comment]
Don't worry just was a path related issue.. I only called it quits because of the time!

**Comment by lxcodes** [comment]
@machuga All set for a review possibly? Made this more "opt-in" instead of completely changing the structure.  Tech debt included: needing to update two sets of instructions, but I'll be down for that.

**Comment by machuga** [comment]
@al3x-edge looks good - we have the "needs review" label too now along with a few others

File: config/database.yml [hunk 1]
```diff
@@ -20,6 +20,10 @@ default: &default
   # For details on connection pooling, see rails configuration guide
   # http://guides.rubyonrails.org/configuring.html#database-pooling
   pool: 5
+  host: db
```
Comment (machuga) [inline config/database.yml]: Let's set these to vars in .env, then create `.env.example-local, .env.example-docker` and set them up appropriately for each env.

File: config/environments/development.rb [hunk 2]
```diff
@@ -39,4 +39,6 @@
 
   # Raises error for missing translations
   # config.action_view.raise_on_missing_translations = true
+
+  config.web_console.whitelisted_ips = '192.168.59.3'
```
Comment (machuga) [inline config/environments/development.rb]: We should likely leave this wide open so this won't break for non-docker devs

Comment (lxcodes) [inline config/environments/development.rb]: :+1:

File: config/database.yml (hunk 1 above)
Comment (lxcodes) [inline config/database.yml]: :+1:

File: config/environments/development.rb (hunk 2 above)
Comment (lxcodes) [inline config/environments/development.rb]: @machuga I think this is actually a Docker specific change.  Noting in web-console documentation:

```
Take a note that IPv4 and IPv6 localhosts are always allowed. This wasn't the case in 2.0.
```

Comment (machuga) [inline config/environments/development.rb]: Ah gotcha

**Comment by lxcodes** [comment]
@machuga Spent some more time on this. Looks pretty solid from both local and docker setups.

**Comment by machuga** [comment]
Aaaaaaand merged :D

**Pull request merged**